docs/content/concepts/flow-lifecycle.md

```diff
@@ -0,0 +1,46 @@
+---
+title: Flow Lifecycle
+sidebar_label: Flow Lifecycle
+sidebar_position: 2
+keywords:
+  - flows
+  - services
+  - discovery
+  - labels
+---
+
+## Flow Lifecycle
+
+To better understand how flows work in Aperture, it is essential to understand
+the lifecycle of a flow. The following diagram shows the lifecycle of a flow in
+Aperture.
+
+<Zoom>
+
+```mermaid
+{@include: ./assets/gen/service/flow-lifecycle.mmd}
+```
+
+</Zoom>
+
+This diagram shows the steps that a flow goes through in Aperture. The steps
+are:
+
+- Selectors: These define scoping rules, identifying and forwarding the flow to
```

Comment:
@tanveergill, I have a small doubt regarding selectors. Since selectors are crucial in determining the flow's path and matching components based on scoping rules, which enables a custom lifecycle for each flow according to its requirements, how does having a fixed lifecycle flow make sense in this case? Specifically, the flow mentioned: Classifier -> FluxMeter -> Sampler -> Rate Limiter -> Schedulers.

Reply:
The matching components are processed in the order of their type:
First, the matching Classifiers, then the matching FluxMeters and so on. The order is based on logical position of each function in the chain. E.g. Sampler is used to statically forbid certain APIs, it should not impact the rate limit. If rate limit is hit first there is not point in waiting for tokens at the Scheduler.